src/main/scala/com/betting/repository/BetRepository.scala

```diff
@@ -1,22 +1,45 @@
 package com.betting.repository
 
-import akka.Done
-import com.betting.BetApplication.system
-import com.betting.domain.Bet
+import com.betting.components.CouchbaseConnectionImpl
+import com.betting.domain.{Bet, BetProtocols}
+import com.couchbase.client.scala.codec.JsonSerializer.CirceEncode
+import com.couchbase.client.scala.kv.{GetResult, MutationResult}
+import com.couchbase.client.scala.query.{QueryOptions, QueryParameters, QueryResult}
+import io.circe.syntax.EncoderOps
 
-import scala.concurrent.Future
+import scala.util.Try
 
-object BetRepository {
-  import system.dispatcher
+object BetRepository extends BetProtocols {
+  val db = new CouchbaseConnectionImpl
 
-  var bets: List[Bet] = Nil
+  def getBet(id: Long): Try[GetResult] = {
+    db.betsCollection.get(id.toString)
+  }
+
+  def placeBet(bet: Bet): Try[MutationResult] = {
+    val docId = bet.id.toString
+    db.betsCollection.upsert(docId, bet.asJson)
+  }
 
-  def getBet(id: Long): Future[Option[Bet]] = Future {
-    bets.find(b => b.id == id)
+  def getBetByAccountId(accountId: Long): Try[QueryResult] = {
```

Comment:
Please check the oficial couchbase documentaytion to see how a QueryResult can be converted to an case class , as TBH `Try[QueryResult]` in not the best output to handle in an application https://docs.couchbase.com/scala-sdk/current/howtos/json.html